src/bucket/test/BucketListTests.cpp

```diff
@@ -155,31 +155,25 @@ basicBucketListTest()
                 {
                     bl.addBatch(
                         *app, i, getAppLedgerVersion(app), {},
-                        LedgerTestUtils::generateValidUniqueLedgerEntries(8),
```

Comment:
On the testing front, I'd recommend adding an invariant check that cache vs database states are consistent (similar to how we do BucketListIsConsistentWithTheDatabase invariant). Given the cost of this check, we can only run it once in a while and only on the nodes that aren't perf-sensitive. But in testing, for example, there's no reason not to enable it on every ledger.

Reply:
If cache population is protocol-agnostic (based on the realization in my other comment), you could run parallel catchup with the cache, then run the new invariant to check state consistency.